### PR TITLE
FINERACT-1151-fix

### PR DESCRIPTION
**Pull request by fynmanoj**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleModelDisbursementPeriod.java [hunk 1]
```diff
@@ -138,4 +138,14 @@ public void setEMIFixedSpecificToInstallmentTrue() {
     public boolean isEMIFixedSpecificToInstallment() {
         return isEMIFixedSpecificToInstallment;
     }
+
+    @Override
+    public BigDecimal rescheduleInterestPortion() {
+        return null;
```
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleModelDisbursementPeriod.java:144]: This looks curious.. is this intentional? Why?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleModelDisbursementPeriod.java [hunk 2]
```diff
@@ -138,4 +138,14 @@ public void setEMIFixedSpecificToInstallmentTrue() {
...
+    }
+
+    @Override
+    public void setRescheduleInterestPortion(BigDecimal rescheduleInterestPortion) {
+        return;
```
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleModelDisbursementPeriod.java:149]: as above

File: fineract-provider/src/main/resources/sql/migrations/core_db/V362__reschedule_interest_to_lrs.sql [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+-- software distributed under the License is distributed on an
+-- "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+-- KIND, either express or implied. See the License for the
+-- specific language governing permissions and limitations
+-- under the License.
+--
+
+
+alter table m_loan_repayment_schedule
+    add column `reschedule_interest_portion` decimal(19,6) DEFAULT NULL after `accrual_interest_derived`;
```
Comment (avikganguly01) [inline fineract-provider/src/main/resources/sql/migrations/core_db/V362__reschedule_interest_to_lrs.sql:22]: Where are the changes to schedule history table?

**Review by avikganguly01 — CHANGES_REQUESTED** [review]
Please include the overlapping long payment holiday multiple reschediling test.

File: fineract-provider/src/main/resources/sql/migrations/core_db/V362__reschedule_interest_to_lrs.sql (hunk 3 above)
Comment (fynmanoj) [reply fineract-provider/src/main/resources/sql/migrations/core_db/V362__reschedule_interest_to_lrs.sql:22]: The history table need not be altered, as the interest portion is carried from the current schedule to the new schedule.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleModelDisbursementPeriod.java (hunk 1 above)
Comment (fynmanoj) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleModelDisbursementPeriod.java:144]: There is one of the two implimentations of the interface "LoanScheduleModelPeriod" (Disbursement period and Repayment Period).  These methods are not significant for Disbursement Period and needed only in Repayment Period, hence they are marked dummy.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleModelDisbursementPeriod.java (hunk 2 above)
Comment (fynmanoj) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleModelDisbursementPeriod.java:149]: There is one of the two implimentations of the interface "LoanScheduleModelPeriod" (Disbursement period and Repayment Period).  These methods are not significant for Disbursement Period and needed only in Repayment Period, hence they are marked dummy.

**Review by avikganguly01 — APPROVED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleModelDisbursementPeriod.java (hunk 1 above)
Comment (avikganguly01) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleModelDisbursementPeriod.java:144]: @vorburger : We can move away from this abstraction and change all the usages in a subsequent PR if you feel it's appropriate. Merging it for now.

**Pull request merged**